frontend/src/components/utils.js

```diff
@@ -11,39 +11,55 @@ export const flatData = (data) => {
 export const pivotHeatMapData = (payload) => {
   // this transformation takes into account that not all criteria have the same rating set
 
-  // create a student lookup for student_id(key) and section_id(value) for later use
-  const studentLookup = payload.denormalized_data.reduce((acc, curr) => (
-    Object.assign(acc, { [curr.student_id]: curr.section_id })), {});
+  const studentSections = {};
+  payload.denormalized_data.forEach((item) => {
+    if (Object.prototype.hasOwnProperty.call(studentSections, item.student_id)) {
+      if (studentSections[item.student_id].indexOf(item.section_id) < 0) {
+        studentSections[item.student_id].push(item.section_id);
+      }
+    } else {
+      studentSections[[item.student_id]] = [item.section_id];
+    }
+  });
 
   // extract the sections for use
-  const sections = payload.sections.map((s) => s.sis_section_id);
+  const sections = payload.sections.map((s) => [
```

Comment:
Would it make sense to create this as an object with named properties? I think it would make the later code more readable, but it's pretty minor, so if there's any reason not to change it that's fine.